pvlib/modelchain.py

```diff
@@ -926,17 +952,65 @@ def prepare_inputs(self, weather, times=None):
             self.weather['temp_air'] = 20
         return self
 
+    def prepare_inputs_from_poa(self, total_irrad):
+        """
+        Prepare the solar position, irradiance, and irradiance inputs to
+        the model, starting with plane-of-array irradiance.
+
+        Parameters
+        ----------
+        total_irrad : DataFrame
+            Irradiance column names must include ``'poa_global'``,
+            ``'poa_direct'`` and ``'poa_diffuse'``.
+
+        Notes
+        -----
+        Assigns attributes: ``solar_position``, ``airmass``, ``aoi``
+
+        See also
+        --------
+        pvlib.modelchain.ModelChain.prepare_inputs
+        """
+
+        req_keys = {'poa_global', 'poa_direct', 'poa_diffuse'}
+        if not req_keys <= set(total_irrad.columns):
+            raise ValueError(
+                "Incomplete irradiance data. Weather data must include "
+                "'poa_global', 'poa_direct' and 'poa_diffuse'.\n"
+                "Detected data: {0}".format(list(total_irrad.columns)))
+
+        self.total_irrad = total_irrad
+
+        self._prep_inputs_solar_pos()
+        self._prep_inputs_airmass()
+
+        # PVSystem.get_irradiance and SingleAxisTracker.get_irradiance
+        # and PVSystem.get_aoi and SingleAxisTracker.get_aoi
+        # have different method signatures. Use partial to handle
+        # the differences.
+        if isinstance(self.system, SingleAxisTracker):
+            self._prep_inputs_tracking()
+        else:
+            self._prep_inputs_fixed()
+
+        if self.weather.get('wind_speed') is None:
+            self.weather['wind_speed'] = 0
+        if self.weather.get('temp_air') is None:
+            self.weather['temp_air'] = 20
```

Comment:
if you move this to immediately after `self._prep_inputs_airmass()` then you have another common sequence with `prepare_inputs`.